Sources/Error.swift

```diff
@@ -50,17 +50,38 @@ public enum QueryError: Error {
         switch self {
         case .invalidSelection(let fieldKeyPath):
             return "Selection for \(fieldKeyPath) is invalid. Make sure it has at most 1 '.' character in it"
-        case .hitSelectionLimit:
+        case .maxSelectionLimitExceeded:
             return "Can select at most 99 key paths when using the select operator on a content type"
+        case .maximumLimitExceeded:
+            return "When limiting the results of a query, results must be limited to a value less than or equal to 1000."
+        case .mimetypeSpecifiedOnEntry:
+            return "Mimetype group can only be specified when querying Assets." +
+                   "The content_type_id parameter of the query should be nil for mimetype_group to work."
+        case .invalidOrderProperty:
+            return "Either 'sys' or 'fields' properties must be specified. Prefix your propety name with 'fields.' or 'sys'"
+        case .specifyContentType:
+            return "A content_type must be specified for this query."
+        case .textSearchTooShort:
```

Comment:
Just a question here, are you validating this stuff to prevent an invalid request from ever being sent? Just considering the possibility that one of these rules change, then you'd have to update the SDK again, no?